File: week12/src/main/java/cstopics/FishAgent.java
package cstopics;

import jade.core.Agent;
import java.awt.Color;

/**
 *
 * @author WillyUgarte
 */
public class FishAgent extends Agent {
	public int x;
	public int y;
	public int status; //1:vigilar,2:comer,3:subir,4:bajar
	public Color color;
	public int size;
	public int speed;

	public FishAgent() {
		x = (int) (100 * Math.random());
		y = (int) (500 * Math.random());
		color = Color.RED;
		size = (int) (25 * Math.random() + 5);
		speed = 10 * 25 / size;
	}

	public void definirStatus() {
		if (y < 500) {
			status = 1;
			if (x > 100 + HostAgent.x_center)
			status = 4;
		}
		else if (y > 900) {
			status = 2;
			if (x < 30 + HostAgent.x_center)
			status = 3;
		}
		else {
			if (x < 30 + HostAgent.x_center)
			status = 3;
			else
			status = 4;
		}
	}

	public void nadar(){
		switch(status){
			case 1: x += speed;
			break;
			case 2: x -= speed;
			break;
			case 3: y -= speed;
			break;
			case 4: y += speed;
			break;
		}
	}
}

File: week12pade/fishagent.py
import random

from globals import Global
from pade.core.agent import Agent

class FishAgent(Agent):
    def __init__(self) -> None:
        self.x = random.randint(1, 100)
        self.y = random.randint(1, 500)
        self.color = random.randint(0, 0xffffff)
        self.size = random.randint(5, 30)
        self.speed = 10 * 25 / self.size
        self.status = -1

    def updateStatus(self):
        if self.y < 500:
            self.status = 1
            if self.x > 100 + Global.x_center:
                self.status = 4;

        elif self.y > 900:
            self.status = 2;
            if self.x < 30 + Global.x_center:
                self.status = 3

        else:
            if self.x < 30 + Global.x_center:
                self.status = 3
            else:
                self.status = 4

    def swim(self):
        if self.status == 1: self.x += self.speed
        elif self.status == 2: self.x -= self.speed
        elif self.status == 3: self.y -= self.speed
        elif self.status == 4: self.y += self.speed


